src/test/java/com/google/edith/ReceiptFileHandlerServiceTest.java

```diff
@@ -0,0 +1,134 @@
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.FileInfo;
+import com.google.appengine.tools.development.testing.LocalBlobstoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.common.collect.ImmutableList;
+import com.google.edith.services.ReceiptFileHandlerService;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public final class ReceiptFileHandlerServiceTest {
+
+  private final LocalServiceTestHelper testHelper =
+      new LocalServiceTestHelper(
+          new LocalBlobstoreServiceTestConfig(), new LocalDatastoreServiceTestConfig());
+
+  private ReceiptFileHandlerService receiptFileHandlerService;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    testHelper.setUp();
+    receiptFileHandlerService = new ReceiptFileHandlerService(blobstoreService);
+  }
+
+  @After
+  public void tearDown() {
+    testHelper.tearDown();
+  }
+
+  @Mock BlobstoreService blobstoreService;
+
+  @Mock HttpServletRequest request;
+
+  @Mock HttpServletResponse response;
+
+  /** Checks if a file was not uploaded in Blobstore. */
+  @Test
+  public void getUploadedFileUrl_ifBlobDidNotUpload_returnEmptyList() throws IOException {
+    Map<String, List<FileInfo>> emptyMap = Collections.emptyMap();
+    when(blobstoreService.getFileInfos(request)).thenReturn(emptyMap);
+
+    List<FileInfo> uploadedFileInfo =
+        receiptFileHandlerService.getUploadedFileUrl(request, "example");
+
+    assertTrue(uploadedFileInfo.isEmpty());
+  }
+
+  /** Checks if a file uploaded successfully in Blobstore returns BlobInfo. */
+  @Test
+  public void getUploadedFileUrl_ifBlobUploaded_returnFileInfo() throws IOException {
+    Date creationDate = new Date();
+    FileInfo uploadFile = new FileInfo("blob", creationDate, "receipt", 0L, "hash", "edith");
+    List<FileInfo> files = ImmutableList.of(uploadFile);
+    Map<String, List<FileInfo>> fileInfos = new HashMap<>();
+    fileInfos.put("fileName", files);
+    when(blobstoreService.getFileInfos(request)).thenReturn(fileInfos);
+
+    List<FileInfo> uploadedFileInfo =
+        receiptFileHandlerService.getUploadedFileUrl(request, "fileName");
+
+    assertFalse(uploadedFileInfo.isEmpty());
```

Comment:
Recommend google-truth http://truth.dev

```java
assertThat(uploadedFileInfo).isEmpty();
```

When using Truth, when the list is not empty this test will fail with the error: `expected list to be empty but it contained: <list of items>` which can be helpful for debugging tests

Reply:
Thanks for the suggestion. I changed all tests to use google-truth